### ISSUE TITLE
suggestion: adjust this package's version number to Zenoh's one

### ISSUE THREAD

**Issue by takasehideki** [biyooon-ex/zenohex#37]
The current version of this hex package (the latest release of this repository) is `v0.2.0`, but the corresponding version of Zenoh is `0.10.1-rc`.
I think it would be nice to adjust these versions because communication between different versions of Zenoh may not be possible, making it easier for users to confirm which version number of Zenohex to use for users. 
What about matching version numbers from the next release? (`0.11.0-rc` I expect)

**Comment by pojiro** [comment]
JFYI,  there are possibilities it will be difficult. Because sometimes we would like to bump up "major.minor.patch"'s patch for our reason, like fixed our Elixir program'sbug or updating document or some thing like that.

**Comment by tshakah** [comment]
A solution could be to use semver's build metadata suffix, e.g. `0.2.0+zenoh-0.11.0-rc`. It isn't particularly pretty, but it does give the required information

**Comment by takasehideki** [comment]
@pojiro thank you so much for giving your thoughts. Yes, my idea certainly makes it harder to patch and bump related to Elixir-side updates/issues. My original proposal turned out to be a bad idea, so I would like to withdraw it.

---
@tshakah thank you so much for giving your idea!! I like it❤️ 
One possible concern is that there may have to write more to mix.exs when users want to use this library in their own projects. However, according to the official documentation, `+` and after can be used to give build information, etc., and are considered equivalent to those without `+`.
https://hexdocs.pm/elixir/1.17.0/Version.html
```
iex()> Version.compare("2.0.1+build0", "2.0.1")
:eq
```

If my understanding is correct, it seems that the user can specify the version of zenohex without changing the notation as usual (i.e., just `{:zenohex, "~> 0.2.0"} in `deps()`), and we can also indicate exactly what information is needed (the corresponding version of Zenoh).

**Comment by takasehideki** [comment]
According to the discussion in this thread, I have tried adding a compatible version after `+` in the version number.  
https://github.com/biyooon-ex/zenohex/releases/tag/v0.2.0%2Bzenoh-0.10.1-rc

Unfortunately, I could not do `mix hex.publish` due to the following error.
```
Publishing package...
[#########################] 100%
Publishing failed
Validation error(s)
  version: build number not allowed
```

I guess the reason is the existence of `v0.2.0` that is treated as the same as the below version number.
I was not able to fully verify what I wanted (especially regarding user experience), but I am satisfied with the results. To conclude, I would like to specify version numbers like this so that they are visible to the compatible Zenoh version in the future.

Again, thank you so much @tshakah @pojiro ❤️

**Comment by s-hosoai** [comment]
I have released version Zenohex v0.3.1.
While release tags and NIF versioning were not an issue with v0.3.1+zenoh-0.11.0, it seems that Hex publish does not allow the use of ‘+’ in version numbers. Therefore, using only 0.3.1 in mix.exs.

**Comment by takasehideki** [comment]
I verified this with @s-hosoai, but it seems that `+` is not accepted by `mix hex.publish`.
https://github.com/hexpm/hexpm/blob/main/lib/hexpm/ecto/changeset.ex

We considered other ideas such as `0.3.0-zenoh0.11.0`, but this would likely compromise the user experience (`{:zenohex,"~> 0.3.0"}` alone would not resolve the dependency). 
Therefore, our conclusion is to give up using `+,`,,,;( 
We need to take care of maintenance README and Releases descriptions.

Note that `mix hex.publish --revert 0.3.1` is used because it is confusing to skip 0.3.0.
